misc/validate_accounting.py

```diff
@@ -309,125 +365,117 @@ def __call__(self, msg):
-            self.neq += 1
-            self.err.write('{} not in indexer\n'.format(niceaddr))
+        microalgos = algod_account['amount-without-pending-rewards']
+        indexerAlgos = indexer_account['amount-without-pending-rewards']
+        if indexerAlgos == microalgos:
+            pass # still ok
         else:
-            xe = ccerr(self.err)
-            indexerAlgos = i2v['amount-without-pending-rewards']
-            if indexerAlgos == microalgos:
-                pass # still ok
+            emsg = 'algod v={} i2 v={} '.format(microalgos, indexerAlgos)
+            if address == reward_addr:
+                emsg += ' Rewards account'
+            elif address == fee_addr:
+                emsg += ' Fee account'
+            xe(emsg)
+        assets = algod_account.get('assets')
+        i2assets = indexer_account.get('assets')
+        if i2assets:
+            if assets:
+                emsg = assetEquality(i2assets, assets)
+                if emsg:
+                    xe('{} {}\n'.format(niceaddr, emsg))
             else:
-                emsg = 'algod v={} i2 v={}'.format(microalgos, indexerAlgos)
-                if address == reward_addr:
-                    emsg += ' Rewards account'
-                elif address == fee_addr:
-                    emsg += ' Fee account'
-                xe(emsg)
-            i2assets = i2v.get('assets')
-            if i2assets:
-                if assets:
-                    emsg = assetEquality(i2assets, assets)
-                    if emsg:
-                        xe('{} {}\n'.format(niceaddr, emsg))
-                else:
-                    allzero = True
-                    nonzero = []
-                    for assetrec in i2assets:
-                        if assetrec.get('amount', 0) != 0:
-                            nonzero.append(assetrec)
-                            allzero = False
-                    if not allzero:
-                        xe('{} indexer has assets but not algod: {!r}\n'.format(niceaddr, nonzero))
-            elif assets:
                 allzero = True
                 nonzero = []
-                for assetrec in assets:
-                    if assetrec.get('amount',0) != 0:
-                        allzero = False
+                for assetrec in i2assets:
+                    if assetrec.get('amount', 0) != 0:
                         nonzero.append(assetrec)
+                        allzero = False
                 if not allzero:
-                    emsg = '{} algod has assets but not indexer: {!r}\n'.format(niceaddr, nonzero)
-                    xe(emsg)
-            i2acfg = i2v.get('created-assets')
-            # filter out deleted entries that indexer is showing to us
+                    xe('{} indexer has assets but not algod: {!r}\n'.format(niceaddr, nonzero))
+        elif assets:
+            allzero = True
+            nonzero = []
+            for assetrec in assets:
+                if assetrec.get('amount',0) != 0:
```

Comment:
you can use the 'deleted' flag for this now, slightly more correct since an asset with 0 balance is still valid.